src/components/UriViewer/LogViewer.ts

```diff
@@ -0,0 +1,224 @@
+};
+
+const modalMaxWidth = 1100;
+const tabMaxWidth = modalMaxWidth / 4 - 20;
+
+export const LogViewer = _.flow(withDisplayName('LogViewer'))(({ modalTitle, logs, onDismiss }: LogViewerProps) => {
+  const [currentlyActiveLog, setCurrentlyActiveLog] = _.isEmpty(logs)
+    ? useState<LogInfo | undefined>(undefined)
+    : useState<LogInfo | undefined>(logs[0]);
+
+  // string = fetched log content, undefined = loading, null = error.
+  const [activeTextContent, setActiveTextContent] = useState<string | undefined | null>(undefined);
+  const [activeDownloadUri, setActiveDownloadUri] = useState<string | undefined>(undefined);
+  const signal = useCancellation();
+  const fetchLogContent = useCallback(
+    async (azureBlobUri: string): Promise<FetchedLogData | null> => {
+      if (!isAzureUri(azureBlobUri)) {
+        return null;
+      }
+      try {
+        const response = await Ajax(signal).AzureStorage.blobMetadata(azureBlobUri).getData();
+        const uri = _.isEmpty(response.azureSasStorageUrl) ? response.azureStorageUrl : response.azureSasStorageUrl;
+        return { textContent: response.textContent, downloadUri: uri };
+      } catch (e) {
+        return null;
+      }
+    },
+    [signal]
+  );
+
+  useOnMount(() => {
+    fetchLogContent(logs[0].logUri).then((content) => {
+      if (_.isEmpty(content?.textContent)) {
+        setActiveTextContent(null);
+      } else {
+        setActiveTextContent(content?.textContent);
+      }
+      setActiveDownloadUri(content?.downloadUri);
+    });
+  });
+
+  useEffect(() => {
+    const fetch = async (logUri: string) => {
+      const res = await fetchLogContent(logUri);
+      if (_.isEmpty(res?.textContent)) {
+        setActiveTextContent(null); // Fetch failed. Display error.
+      } else {
+        setActiveTextContent(res?.textContent);
+      }
+      setActiveDownloadUri(res?.downloadUri);
+    };
+
+    // when switching tabs, reset content to undefined while we fetch the new content.
+    setActiveTextContent(undefined);
+    setActiveDownloadUri(undefined);
+
+    // tab switching set the currently active log (which triggers this effect). Fetch the content for the new log.
+    const uri = currentlyActiveLog?.logUri;
+    if (!_.isEmpty(uri)) {
+      fetch(uri);
+    }
+  }, [logs, currentlyActiveLog, fetchLogContent]);
+
+  const renderActiveTextContent = () => {
+    if (activeTextContent === undefined) {
+      return div([centeredSpinner()]);
+    }
+    if (activeTextContent === null) {
+      return div([
+        "Log file could not be loaded. If the workflow or task is still in progress, the log file likely hasn't been generated yet. Some logs may be unavailable if the workflow or task failed before they could be generated.",
+      ]);
+    }
+    return div([activeTextContent]);
+  };
+
+  const tabsArray: SimpleTabProps[] = logs.map((log) => {
+    return { key: log.logKey, title: log.logTitle, width: tabMaxWidth } as SimpleTabProps;
```

Comment:
```suggestion
    return { key: log.logKey, title: log.logTitle, width: tabMaxWidth };
```

Is the cast necessary here? It doesn't look like it should be.